### PR TITLE
Add Checkspell CI job

### PR DESCRIPTION
**Pull request by xnameTM**
## Summary
This PR includes Checkspell CI job to repo.

## Test plan
Check CI jobs... 👉👈

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        steps:
+            - uses: actions/checkout@v2
+            - uses: streetsidesoftware/cspell-action@v6
+              with:
+                files: '**/*'
+                root: '.'
+                config: './cspell.json'
+                inline: warning
+                strict: true
+                use_cspell_files: false
```
Comment (tjzel) [inline .github/workflows/spellcheck.yml:22]: newline 😎

File: cspell.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "workletized",
+        "barfoo",
+        "devs"
+    ],
+    "ignorePaths": [
+        "node_modules",
+        "docs/src/components/Testimonials"
+    ],
+    "useGitignore": true
+}
```
Comment (tjzel) [inline cspell.json:18]: newline 😎

**Review by tjzel — APPROVED** [review]
:shipit:

**Review by tomekzaw — CHANGES_REQUESTED** [review]

File: .github/workflows/spellcheck.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: streetsidesoftware/cspell-action@v6
+        with:
+          files: '**/*'
+          root: '.'
+          config: './cspell.json'
+          inline: warning
+          strict: true
+          use_cspell_files: false
```
Comment (tomekzaw) [inline .github/workflows/spellcheck.yml:22]: Let's add a newline at the end of the document

File: .github/workflows/spellcheck.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: 'Check spelling'
+on:
+  push:
+    branches:
+      - main
```
Comment (tomekzaw) [inline .github/workflows/spellcheck.yml:5]: Let's add like in other `.yml` files
```
  merge_group:
    branches:
      - main
```

File: cspell.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (tomekzaw) [inline cspell.json:1]: Let's use 2 spaces here as well

**Review by tomekzaw — CHANGES_REQUESTED** [review]
Left some minor suggestions in the comments

File: .github/workflows/spellcheck.yml (hunk 4 above)
Comment (xnameTM) [reply .github/workflows/spellcheck.yml:8]: ok

**Review by tomekzaw — APPROVED** [review]
Thanks!

**Pull request merged**